src/main/java/roomescape/controller/ReservationController.java

```diff
@@ -13,35 +11,45 @@
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestController;
+import roomescape.controller.dto.ReservationCreateRequest;
+import roomescape.controller.dto.ReservationCreateResponse;
+import roomescape.controller.dto.ReservationFindResponse;
 import roomescape.domain.Reservation;
+import roomescape.service.ReservationService;
 
 @RestController
 @RequestMapping("/reservations")
 public class ReservationController {
 
-    private final List<Reservation> reservations = new ArrayList<>();
-    private final AtomicLong index = new AtomicLong(1);
+    private final ReservationService reservationService;
+
+    public ReservationController(ReservationService reservationService) {
+        this.reservationService = reservationService;
+    }
```

Comment:
의존성 주입에는 3가지 방법이 있습니다. 
1. 생성자 주입
2. 필드 주입
3. setter 주입

필드 주입은 필드가 `public`으로 열려있지 않는 한 다른 객체를 의존하도록 변경할 수 없으니 제외합니다.
또한 setter를 사용하려면 필드의 재할당이 가능해 지기 때문에 객체의 생성 및 의존성 주입이 완료된 이후에도 언제든지 의존성의 변경이 일어날 수 있다고 생각해요.

위와 같은 이유로 생성자 주입을 사용하게 되었습니다.

Reply:
spring에서 field injection 시 field가 public이어야할까요?
spring은 객체를 어떤 방식으로 관리할까요?

Reply:
> 필드 주입은 필드가 public으로 열려있지 않는 한 다른 객체를 의존하도록 변경할 수 없으니 제외합니다.

말이 조금 모호했네요. 제가 전달하려던 메시지는 "테스트와 같은 상황에서 개발자가 직접 의존관계를 주입해야 하는 경우, 필드를 `public`으로 열어 주어야 한다." 였습니다.

스프링이 제공하는 field injection은 `private` 필드에 `@Autowired`를 붙여주는 것 만으로도 잘 동작하는 것으로 알고 있습니다.

---

추가 학습이 가능할 것 같아 몇 가지를 "간단히" 알아보았는데요.

1. 여러 이유로 field injection은 권장되지 않네요.

앞선 코멘트에서 언급한 것 처럼 field injection을 사용하게 되면 해당 필드의 불변성을 제공할 수가 없습니다.
또한, 필드의 초기화가 객체의 생성 단계에서 완료되지 않기 때문에 NPE가 발생될 위험도 있다고 하네요!

이외에도 다양한 이유로 field injection을 권장하지 않는 것 같습니다.

[baeldung - Why Is Field Injection Not Recommended?](https://www.baeldung.com/java-spring-field-injection-cons)
[What exactly is Field Injection and how to avoid it?](https://stackoverflow.com/questions/39890849/what-exactly-is-field-injection-and-how-to-avoid-it)

(사실, 의존성을 직접 주입할 수 있다는 점 이외에는 대부분의 단점을 setter injection 역시 공통적으로 가지는 것 같은데요. 이에 대해서는 추가 학습이 필요할 것 같습니다.)

2. 빈의 생명 주기

빈의 생명 주기가 궁금해져서 "간단히" 알아보았는데요. [medium - Spring Bean Lifecycle Full Guide](https://medium.com/@TheTechDude/spring-bean-lifecycle-full-guide-f865966e89ce) 
기본적으론 스프링 빈이 singleton으로 관리되기 때문에 Application이 실행될 때 빈을 모두 생성해서 저장해 둔다고 합니다.

field injection은 reflection을 사용한다고 하니 모든 생성과 초기화 단계가 구분되어 진행될 것 같아요.
[medium - spring boot dependency injection](https://medium.com/@reetesh043/spring-boot-dependency-injection-137f85f84590)

이 이상의 깊이로 학습하는 것은 지금 단계에서의 영역이 아닌 것 같아 살포시 발을 빼 볼까 합니다...ㅎㅎ

Reply:
스프링 빈은 어디서 관리할까요?

Reply:
> 스프링 빈이 singleton으로 관리되기 때문에 Application이 실행될 때 빈을 모두 생성해서 저장해 둔다고 합니다.

스프링 빈은 어디서 관리할까요?

Reply:
이 부분은 조금 더 깊게 학습해 볼 수 있을 만한 키워드인 것 같아서 여러 문서들을 찾아보았습니다!

간단하게 요약하면, Spring 어플리케이션이 시작할 때, Spring Container가 내부에 Spring Bean을 생성해 저장합니다. 
[spring docs - Introduction to the Spring IoC Container and Beans](https://docs.spring.io/spring-framework/reference/core/beans/introduction.html)

### Spring Container, Application Context, Bean Factory

 흔히 Spring에서 사용하는 위의 용어들이 헷갈려서 각 용어들을 학습해 보았는데요. Spring in Action 에서 세 용어의 관계를 찾을 수 있었습니다.
 
 Spring Container 는 어플리케이션 실행에 필요한 Spring bean의 생명 주기를 관리하는 객체입니다. 그리고, Spring Container를 크게 Application Context 와 Bean Factory 로 나눌 수 있으며 Application Context가 Bean Factory 보다 훨씬 많은 기능을 제공해 주기에 대부분의 경우 Application Context 를 사용한다고 하네요!